src/modules/Dropdown/DropdownSearchInput.js

```diff
@@ -0,0 +1,114 @@
+import cx from 'classnames'
+import _ from 'lodash'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+
+import {
+  createShorthandFactory,
+  customPropTypes,
+  META,
+  getUnhandledProps,
+} from '../../lib'
+
+/**
+ * A search item sub-component for Dropdown component.
+ */
+class DropdownSearchInput extends Component {
+  static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** A ref handler for input. */
+    inputRef: PropTypes.func,
+
+    /** A name of the input. */
+    name: PropTypes.oneOfType([
+      PropTypes.number,
+      PropTypes.string,
+    ]),
+
+    /** Custom styles. */
+    style: PropTypes.object,
+
+    /** An input can receive focus. */
+    tabIndex: PropTypes.oneOfType([
+      PropTypes.number,
+      PropTypes.string,
+    ]),
+
+    /** The HTML input type. */
+    type: PropTypes.string,
+
+    /** Stored value. */
+    value: PropTypes.oneOfType([
+      PropTypes.number,
+      PropTypes.string,
+    ]),
+
+    /** A width of an input. */
+    width: PropTypes.oneOfType([
```

Comment:
I would prefer we just use the `width` property within the `style` prop instead of breaking it out into a separate prop.